### PR TITLE
wlr-keyboard: keep track of number of keycodes pressed

### PR DESCRIPTION
**Pull request by acrisci**
Makes this useful for sway bindcode.

### CODE REVIEW

**Comment by SirCmpwn** [comment]
Thanks!

File: types/wlr_seat.c [hunk 1]
```diff
@@ -874,11 +874,9 @@ void wlr_seat_keyboard_enter(struct wlr_seat *seat,
 
 		struct wl_array keys;
 		wl_array_init(&keys);
-		for (size_t i = 0; i < WLR_KEYBOARD_KEYS_CAP; ++i) {
-			if (keyboard->keycodes[i] != 0) {
-				uint32_t *p = wl_array_add(&keys, sizeof(uint32_t));
-				*p = keyboard->keycodes[i];
-			}
+		for (size_t i = 0; i < keyboard->num_keycodes; ++i) {
+			uint32_t *p = wl_array_add(&keys, sizeof(uint32_t));
```
Comment (martinetd) [inline types/wlr_seat.c]: wl_array_add can return null

Comment (emersion) [reply types/wlr_seat.c]: Can you grep for wl_array_add and make sure that all return values are checked?

Comment (emersion) [reply types/wlr_seat.c]: Can you grep for wl_array_add and make sure that all return values are checked?

Comment (martinetd) [reply types/wlr_seat.c]: see top comment, about half are. I'll submit a PR to fix all that don't I guess.

**Pull request merged**